### ISSUE TITLE
Iray

### ISSUE THREAD

**Issue by Astuceman** [Neon22/DazStudio-Light-Panel#1]
Your script is amazing but some feature specific to iray lights are missing.
Can you add it ?

<img width="732" alt="capture 36" src="https://user-images.githubusercontent.com/19632209/40277107-4a2c2876-5c22-11e8-805b-58d0c76fc981.png">

**Comment by Neon22** [comment]
I haven't used this in the new Iray system. How much is it working ? I expect the API to have changed a lot and maybe it doesn't work at all anymore ?

**Comment by Astuceman** [comment]
Everything in Master control is working
I've tried with each type of light (you can see in screenshot)

Soft and bias are not working with Iray (maybe with 3delight but I have'nt test it)

it only miss the new feature

**Comment by Neon22** [comment]
I will try to get to it over next few days but not tried to find Iray params before in Daz. Not sure where the docs are for Iray/Daz..

**Comment by Astuceman** [comment]
I'm not an expert in javascript so ... I'm not sure that i can help you

but ... here what I have found http://docs.daz3d.com/doku.php/public/software/dazstudio/4/referenceguide/scripting/start
http://docs.daz3d.com/doku.php/public/software/dazstudio/4/referenceguide/scripting/api_reference/start

**Comment by Neon22** [comment]
yes . Softness and bias are only used by 3Dlight style biased renderers. Iray is unbiased and physically accurate.
I see that Area is now a boolean but there is no indication of thenew fields you highlight in the docs :( here:
- http://docs.daz3d.com/doku.php/public/software/dazstudio/4/referenceguide/scripting/api_reference/object_index/light_dz
Even under materials there is no link to the Iray materials. I wil have to look in the forum and possibly ask questions...
- http://docs.daz3d.com/doku.php/public/software/dazstudio/4/referenceguide/scripting/api_reference/object_index/material_dz

**Comment by Neon22** [comment]
enquiring
- https://www.daz3d.com/forums/discussion/253776/iray-light-params#latest

**Comment by Astuceman** [comment]
I have found this

- http://docs.daz3d.com/doku.php/public/software/dazstudio/4/referenceguide/interface/panes/surfaces/shaders/iray_uber_shader/start

**Comment by Astuceman** [comment]
I have published this on deviantart 

<img width="490" alt="capture 33" src="https://user-images.githubusercontent.com/19632209/40294129-f2ba429c-5cdc-11e8-81a9-e9f1ec5ec09f.PNG">

And people really like it
Some stats

<img width="200" alt="capture 39" src="https://user-images.githubusercontent.com/19632209/40294157-15fbe058-5cdd-11e8-8fe4-cfed928b20bc.PNG">

Check out here http://fav.me/dcc0xsw
Many people asking for Iray support

**Comment by Neon22** [comment]
OK. got tip from Rob on forum. Looks good.
Hoping to find time to get to this soon. Just working on a pyglet project for someone else right now.
